### PR TITLE
feat: caching

### PR DESCRIPTION
**Pull request by sabotagebeats**
adding caching to action.yml

### CODE REVIEW

**Comment by sabotagebeats** [comment]
this might not be possible? https://stackoverflow.com/questions/58611841/how-can-i-reference-other-actions-from-my-github-actions-action-yml-file

**Comment by sabotagebeats** [comment]
https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

**Comment by sabotagebeats** [comment]
going to try building a shell script

File: action.yml [hunk 1]
```diff
@@ -3,8 +3,35 @@ author: 'ApeWorX Ltd.'
...
+  
+  # todo: cache pip
+  # - uses: actions/cache@v2
+
+  # todo: select version of ape via input  
+  - run: pip install eth-ape
+    shell: bash
+
+  # todo: allow selecting plugins
+  - run: ape plugins install solidity vyper
```
Comment (unparalleled-js) [inline action.yml:36]: Can we use the project's config file to determine which compilers are needed before installing them?

Comment (sabotagebeats) [reply action.yml]: we just resolved this :)

**Pull request merged**